src/hotspot/os/linux/os_linux.cpp

```diff
@@ -378,22 +378,40 @@ static void next_line(FILE *f) {
   } while (c != '\n' && c != EOF);
 }
 
-void os::Linux::kernel_version(long* major, long* minor) {
+void os::Linux::kernel_version(long* major, long* patch, long* sub) {
   *major = -1;
-  *minor = -1;
+  *patch = -1;
+  *sub   = -1;
 
   struct utsname buffer;
   int ret = uname(&buffer);
   if (ret != 0) {
     log_warning(os)("uname(2) failed to get kernel version: %s", os::errno_name(ret));
     return;
   }
-  int nr_matched = sscanf(buffer.release, "%ld.%ld", major, minor);
-  if (nr_matched != 2) {
-    log_warning(os)("Parsing kernel version failed, expected 2 version numbers, only matched %d", nr_matched);
+  int nr_matched = sscanf(buffer.release, "%ld.%ld.%ld", major, patch, sub);
```

Comment:
Are we sure the kernel version always have 3 components? It would be safer to handle 2-component version as well, and set the third component to `0` in that case.

Reply:
I took another look; technically, only one is guaranteed by the source.
However, in practice, there are always three (checked in Linux v3).

Reply:
OK. I misread the code and thought the error is fatal when some components are not defined. But it is only a warning, so we are probably fine. Do you think it would be reasonable to initialize the components to `0`, though, so on 1- or 2-component version we would not have e.g. `2.18.-1`?

Reply:
Yes, I think that is much better, thanks!